rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbIntegrationTest.java

```diff
@@ -0,0 +1,567 @@
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(Lifecycle.PER_CLASS)
+class FeePerKbIntegrationTest {
+    private final FeePerKbConstants feePerKbConstants = FeePerKbMainNetConstants.getInstance();
+    private FeePerKbSupport feePerKbSupport;
+    private SignatureCache signatureCache;
+    private Coin currentFeePerKb;
+    private Coin differentFeePerKbVote;
+
+    @BeforeAll
+    void setUp() {
+        StorageAccessor inMemoryStorage = new InMemoryStorage();
+        FeePerKbStorageProvider feePerKbStorageProvider = new FeePerKbStorageProviderImpl(inMemoryStorage);
+        feePerKbSupport = new FeePerKbSupportImpl(feePerKbConstants, feePerKbStorageProvider);
+        signatureCache = mock(SignatureCache.class);
+        differentFeePerKbVote = Coin.valueOf(60_000L);
+    }
+
+    /**
+     * Without any previous voting.
+     */
+    @Test
+    @Order(0)
+    void genesisFeePerKb() {
+        Coin genesisFeePerKb = feePerKbConstants.getGenesisFeePerKb();
+        currentFeePerKb = genesisFeePerKb;
+        assertFeePerKbValue(genesisFeePerKb);
+    }
+
+    /**
+     * Authorizer 1 and 2 vote the same value
+     */
+    @Test
+    @Order(1)
+    void firstVoting() {
+        Coin firstFeePerKbVote = Coin.valueOf(50_000L);
+        // Send 1 vote from an authorizer
+        voteFeePerKb(firstFeePerKbVote, FeePerKbVoteCaller.FIRST_AUTHORIZED.getRskAddress());
+        // Get fee per kb, shouldn't have changed because only one vote was added
+        assertFeePerKbValue(currentFeePerKb);
+
+        // Second vote from a different authorizer
+        voteFeePerKb(firstFeePerKbVote, FeePerKbVoteCaller.SECOND_AUTHORIZED.getRskAddress());
+        currentFeePerKb = firstFeePerKbVote;
+
+        // Get fee per kb, should return the newly voted value
+        assertFeePerKbValue(firstFeePerKbVote);
+    }
+
+    /**
+     * Authorizers 2 and 3 vote the same value
+     */
+    @Test
+    @Order(2)
+    void secondVoting() {
+        Coin secondFeePerKbVote = Coin.valueOf(100_000L);
+        // First vote for a new value
+        voteFeePerKb(secondFeePerKbVote, FeePerKbVoteCaller.SECOND_AUTHORIZED.getRskAddress());
+        // Get fee per kb, should return the first value voted because the new value has only 1 vote
+        assertFeePerKbValue(currentFeePerKb);
+
+        // Second vote for the same value, different authorizer
+        voteFeePerKb(secondFeePerKbVote, FeePerKbVoteCaller.THIRD_AUTHORIZED.getRskAddress());
+        currentFeePerKb = secondFeePerKbVote;
+
+        // Get fee per kb, should return the second round vote value
+        assertFeePerKbValue(secondFeePerKbVote);
+    }
+
+    /**
+     * Authorizers 1 and 2 vote different values. Then authorizer 3 votes the same value as
+     * authorizer 1
+     */
+    @Test
+    @Order(3)
+    void thirdVoting() {
```

Comment:
i would make the test names more descriptive instead of having the comments above them.
wdyt?

Reply:
Totally agree, @julia-zack!

Reply:
Done! @jeremy-then @julia-zack @marcos-iov